ui/src/admin/components/chronograf/AllUsersTable.js

```diff
@@ -0,0 +1,208 @@
+import React, {Component, PropTypes} from 'react'
+
+import uuid from 'node-uuid'
+
+import AllUsersTableHeader from 'src/admin/components/chronograf/AllUsersTableHeader'
+import AllUsersTableRowNew from 'src/admin/components/chronograf/AllUsersTableRowNew'
+import AllUsersTableRow from 'src/admin/components/chronograf/AllUsersTableRow'
+
+import {ALL_USERS_TABLE} from 'src/admin/constants/chronografTableSizing'
+const {
+  colOrganizations,
+  colProvider,
+  colScheme,
+  colSuperAdmin,
+  colActions,
+} = ALL_USERS_TABLE
+
+class AllUsersTable extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isCreatingUser: false,
+    }
+  }
+
+  handleUpdateAuthConfig = fieldName => updatedValue => {
+    const {
+      actionsConfig: {updateAuthConfigAsync},
+      authConfig,
+      links,
+    } = this.props
+    const updatedAuthConfig = {
+      ...authConfig,
+      [fieldName]: updatedValue,
+    }
+    updateAuthConfigAsync(links.config.auth, authConfig, updatedAuthConfig)
+  }
+
+  handleAddToOrganization = user => organization => {
+    // '*' tells the server to fill in the current defaultRole of that org
+    const newRoles = user.roles.concat({
+      organization: organization.id,
+      name: '*',
+    })
+    this.props.onUpdateUserRoles(
+      user,
+      newRoles,
+      `${user.name} has been added to ${organization.name}`
+    )
+  }
+
+  handleRemoveFromOrganization = user => role => {
+    const newRoles = user.roles.filter(
+      r => r.organization !== role.organization // TODO: the organization upstream of this should use .id instead of .organization for this property
```

Comment:
Has this TODO been done?

Reply:
We have decided to remove this TODO because roles are structured in such a way that they do not have IDs because they are nested under the User resource. We can revisit this if we ever establish roles as their own resource.